pkgs/development/libraries/gtk/2.x.nix

```
@@ -61,6 +61,10 @@ stdenv.mkDerivation rec {
     ++ optionals cupsSupport [ cups ]
     ++ optionals stdenv.isDarwin [ AppKit Cocoa ];
 
+  preConfigure = lib.optionalString (lib.versionAtLeast stdenv.hostPlatform.darwinMinVersion "11") ''
```

to avoid mass linux rebuild
```suggestion
  preConfigure = lib.optionalString (lib.versionAtLeast stdenv.hostPlatform.darwinMinVersion "11" && stdenv.isDarwin) ''
```

---

Amended.

---

Empty string or list don't work that way.  `null` would work, but I'm afraid such `if` would be slightly less readable.

---

Should I drop it, keep as is, or do something else?

---

oh right. @vcunat is right